### ISSUE TITLE
alternate sound(s)

### ISSUE THREAD

**Issue by bburdette** [thetwom/toc2#84]
For a metronome sound, I'd like something plainer.  

- no reverb or echo
- no perceptible pitch

I don't want something that sounds like a tone, say 'G' when I'm in G#.  The note with a diagonal line through it has less pitch than the high note, but it has reverb.  The bass note is too low.  I'd prefer a dry, precise click.   

Side issue: I found the sound choosing interface to be unintuitive.  I couldn't figure out how to choose a sound at all, until I read [this issue](https://github.com/thetwom/toc2/issues/9).  Then I discovered that you click on the note to choose the click sample.  I was thinking of the note as being more about timing, like for people that want triplet feel or want to maybe use a midi file or something.  I see that interface has a lot of flexibility in terms of making a rhythmic 'phrase'.  I am a more simple user that just wants a click and that's it, so I didn't press the note and discover its dual use as sample selection.  

Positive feedback:

The 'tap in' is great - averaging a sequence of taps rather than just using the last two.  Way more precise than the 'just two' method.

Also like the dial interface - allows big adjustments or precise small adjustments.

**Comment by thetwom** [comment]
Thanks a lot for the feedback. 

Sounds: 
- I guess a lot of improvement can be done here, depending on the use case. There was a bit of discussion also here: #69 
- I have the feeling that pitch or reverb is sometimes preferred by people since it helps to be more dominant. 

Maybe you can suggest sounds? It doesn't have to be a finished sound, but maybe a link to a video or similar ...

However, it will definitely take some time, until I find some time to further deal with new sounds. 

Regarding the side issue: It is always good to have feedback of such things. UI design seems to be extremely difficult when doing it for a wide range of people. I would think that clicking on a note to change the tone as well as duration is the most obvious choice since you click on the item you actually want to change. But as you say, it might not be obvious to everyone. And currently the notes are not highlighted to be clickable (like having a different color and not black). In the end, it seems to be like it is a design choice between a very simple interface (e.g. define triplet feel with highlighting the first note) and more flexible as it is now. But maybe there are better ideas.

**Comment by bburdette** [comment]
Well, [here](https://www.zknotes.com/file/890e4cb1-9e27-42cc-8483-d5c9ab34a2fc) is a click sound I recorded from my Boss DB-90 metronome.  Maybe not the best sound ever but ok.  A mechanical clock tick might be similar and less copyright encumbered.   

Re assigning different sounds to different notes.  My first thought is to associate different notes with different pitches of the same sound, not different timbres altogether.  But its probably intuitive if one is used to reading for drums, like maybe an A is open hi hat and A# is closed hi hat, something like that?  Maybe dual labeling of that area would help, although whats an icon that stands for timbre?  A tiny spectrogram?  A sound wave?  A drum?  Or maybe a text label, though there is an appeal to having an entirely text free interface.  

Anyway for myself its totally fine now that I know how it works.

**Comment by davidpiano** [comment]
My response to the question about sounds:

The tick of a mechanical metronome, or something fairly similar. Having no noticeable pitch or tone, probably because it evenly covers a wide range of frequency. Loud. (It's better to need to turn down the volume, than to have a metronome that is useless because it's quiet.)

I think people designing electronic metronomes often choose sounds that are musically interesting or that sound nice, and that's exactly what I DON'T want. A nice tone in the metronome _reduces_ its usability, because that tone interferes with the music.

There's nothing wrong with providing nice metronome sounds for people who like them, as long as you first provide plain mechanical-sounding ones.

**Comment by thetwom** [comment]
I really hope to find some time to look a bit more into sounds in not too far future, for now I guess it has to stay a discussion.

I have not real facts about this, just made some observations, what sounds people would choose, and I am trying to understand why. Of course, one of the main reasons is that it can easily be heard over all the other instruments. My impressions are:
- A short tick is precise but can be quickly overheard since it so short. 
- Yes, these ticks that e.g. mechanical metronomes have, trigger a large range of frequencies. But not always this is an advantage, because a large range of frequencies might be covered by other instruments playing, so the tick does not sound so dominant.
- In contrast, pure frequencies can be very dominant if the frequency is e.g.  well above the frequencies of the playing instruments. But it also can be very annoying :-).
- What also helps seem to be sounds with some kind of reverb. The timing is not as precise anymore, but if the sound plays at the same time as a percussion instrument,  a short signal might get lost. 

So, at the moment I have the impression that there is no ideal sound, just pros and cons and it really depends on the use case. I was once trying to make a sound generator, where one could e.g. try to make signals to contain more like a single frequencies or be more like a click. But here we are back to what sounds nice. I don't think this issue can be ignored, since it is not only about dominance about the sound, but it should not annoy you :-). And artificially created sound did sound not as good in my ears as "real-world" sounds, but maybe this can be improved by investing a bit more time ... 

But all this is just an impression without any significant evidence. So, keep up posting your preference, this really helps to get a better picture.

**Comment by bigboipete** [comment]
@thetwom You are talking about a solution to generate sounds within the app. But as you said, taste makes this solution almost impossible.
I would rather expect an option to select click sounds/samples, that are provided on the device and by the user (like custom ringtones). I don't know if this approach would ease the task technically, but it would free you from the burden to generate the ultimate click sound for everyone.

**Comment by thetwom** [comment]
Yes, I guess, the option to allow the user to use own sounds is something which would be a helpful to some people. It might sound a bit simpler than it finally is. In the end, what is needed:
- decoding of audio codecs: In the past there were some phones, which showed strange behaviour here, why I removed any decoding functionality (which was used for the internal sound initially), but in the requested, decoding would not be needed  for the default sounds. So it might be ok to loose a bit of robustness here ...
- resampling: In order to get optimal audio performance, you must provide the samples at the native sample rate of the phone.

Up to now, the app used different ways to play sounds in the metronome and for "preview", which made all this more complicated. I did move away from that for the [successor app](https://codeberg.org/thetwom/metronome), which should make things simpler. However, there are still a few details missing there, especially much requested features like automatic change of bpm, ..., but things are progressing slowly ... lets see where it gets ...
